app/assets/stylesheets/common/base/sidebar.scss

```diff
@@ -104,8 +102,24 @@
       font-weight: bold;
     }
 
-    .badge-wrapper {
-      font-size: 100%;
+    .sidebar-section-link-content-badge {
+      width: 30%;
+      text-align: right;
+    }
+
+    .sidebar-section-link-content-text {
+      width: 70%;
+
+      .badge-wrapper {
+        font-size: 100%;
+        width: 100%;
+
+        .category-name {
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
```

Comment:
btw. we have a mixin for this (https://github.com/discourse/discourse/blob/main/app/assets/stylesheets/common/foundation/mixins.scss#L60-L60) but nobody remembers it (incl. usually me 😛) so I kinda question its value 😅